citadel_workspace_service/src/kernel/mod.rs

```diff
@@ -381,4 +427,137 @@ mod tests {
             panic!("Wrong packet type");
         }
     }
+    // test
+    #[tokio::test]
+    async fn message_test() -> Result<(), Box<dyn Error>> {
+        citadel_logging::setup_log();
+        info!(target: "citadel", "above server spawn");
+        let bind_address_internal_service: SocketAddr = "127.0.0.1:55568".parse().unwrap();
+
+        // TCP client (GUI, CLI) -> internal service -> empty kernel server(s)
+        let (server, server_bind_address) = citadel_sdk::test_common::server_info_reactive(
+            |conn, _remote| async move {
+                let (sink, mut stream) = conn.channel.split();
+
+                while let Some(_message) = stream.next().await {
+                    let send_message = InternalServiceResponse::MessageSent { cid: conn.cid };
+                    send_back(sink.clone(), send_message).await.unwrap();
```

Comment:
No need for a send_back function. Just call sink.send(send_message)

Reply:
We shouldn't be serializing these packets here to `InternalServiceResponse`. We are inside the server, not the internal service, thus, saying it is an internal service response is misleading.